source/Cosmos.HAL2/PS2Mouse.cs

```diff
@@ -166,8 +169,15 @@ public void HandleMouse(ref INTs.IRQContext context)
 
                 if (HasScrollWheel)
                 {
-                    var xScrollWheelByte = mouseByte[3] & 0x0F;
-                    xScrollWheel = (xScrollWheelByte & 0b1000) == 0 ? xScrollWheelByte : xScrollWheelByte | ~0x0F;
+                    var xScrollWheelByte = mouseByte[3];
+
+                    xScrollWheel = xScrollWheelByte & 0b0000_0111;
+
+                    // If the 4th bit is set, the scroll wheel is negative
+                    if ((xScrollWheelByte & (1 << 4)) == 1 << 4) {
+                        // if we just negate the value it starts at -8 and goes to 0, but we want it to start at 0 and go to -8
+                        xScrollWheel = -(-xScrollWheel + 8);
+                    }
```

Comment:
This doesn't look right. The 4th bit can be the 4th button state, in 5 buttons mode ([OSDev ref](https://wiki.osdev.org/PS/2_Mouse#5_buttons)). Also, you wrote a very similar logic in a more complicated way, this is just a sign extension.

Reply:
You are right, I tried a bunch of things until I found out that it just didnt read the 4th byte at all, I am gonna revert that change.